src/lib/graphqlErrorHandler.ts

```diff
@@ -63,46 +74,56 @@ const reportErrorToSentry = (
   })
 }
 
+type WriteablePartial<T> = { -readonly [P in keyof T]+?: T[P] }
```

Comment:
I know, right?

There's a bit more documentation [here](https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#improved-control-over-mapped-type-modifiers). In short, we're mapping over all the properties in the given object structure and removing `readonly` modifiers with the `-` sign and adding the optional (`?`) modifier with the `+` sign.

Reply:
Interesting!

I'm understanding why this is the desired typing for `result`, and also marveling at the actual type declaration!

Reply:
This is just crazy! This line:)

Reply:
Yeah it’s definitely not the easiest 😄, but I trust we're at the point that we can introduce some of these harder ones when _really_ useful and everybody still being able to follow along, given some notes. (And to be clear, this applies to me too, this comment is as much for others as it is for me.)